worker/src/request.rs

```diff
@@ -0,0 +1,11 @@
+use edgeworker_sys::Request as EdgeRequest;
+use wasm_bindgen::JsValue;
+
+use crate::Request;
+
+pub fn new_worker_req(event_type: String, url: &str) -> Result<Request, JsValue> {
```

Comment:
This should be implemented as a constructor for `Request`. In Rust constructors are written as a `new` method for the struct, e.g.:

```
impl Request {
  pub fn new(url: &str) -> Request {
  ...
  }
}
```

Also the event_type should be automatically set to `fetch` in this case instead of requiring it as a parameter to the constructor.

Reply:
done